spec/controllers/wishlists_controller_spec.rb

```diff
@@ -1,11 +1,23 @@
 require "rails_helper"
 
 RSpec.describe WishlistsController do
-  context "when user is logged in" do
-    before { sign_in_user }
-    let(:wishlist) { create(:wishlist, user: controller.current_user) }
+  let(:wishlist) { create(:wishlist, user: controller.current_user) }
 
-    describe "GET #index" do
+  shared_examples "redirects to the 'sign_in' page" do
+    it "redirects to the 'sign_in' page" do
+      expect(response).to redirect_to(new_user_session_path)
+    end
+  end
+
+  shared_examples "redirects to the wishlists index page" do
+    it "redirects to the wishlists index page" do
+      expect(response).to redirect_to(wishlists_path)
+    end
+  end
```

Comment:
You could make this a one `shared_example` like so:
```ruby
shared_examples "redirects to" do |path_helper|
  it "redirects to the #{path_helper}" do
    expect(response).to redirect_to(send(path_helper))
  end
end
```
i wywoływać to później w  ten sposób:
```ruby
include_examples "redirects to", :new_user_session_path
```